pkg/forwarder/internal/retry/transaction_retry_queue.go

```diff
@@ -3,84 +3,89 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-present Datadog, Inc.
 
-package forwarder
+package retry
 
 import (
 	"fmt"
 	"sync"
 
 	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/forwarder/transaction"
 	"github.com/DataDog/datadog-agent/pkg/util/filesystem"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 	"github.com/hashicorp/go-multierror"
 )
 
-type transactionStorage interface {
-	Serialize([]Transaction) error
-	Deserialize() ([]Transaction, error)
+// TransactionSerializer is an interface to serialize / deserialize transactions
+type TransactionSerializer interface {
+	Serialize([]transaction.Transaction) error
+	Deserialize() ([]transaction.Transaction, error)
 }
 
-type transactionPrioritySorter interface {
-	Sort([]Transaction)
+// TransactionPrioritySorter is an interface to sort transactions.
+type TransactionPrioritySorter interface {
+	Sort([]transaction.Transaction)
 }
 
-// transactionContainer stores transactions in memory and flush them to disk when the memory
+// TransactionRetryQueue stores transactions in memory and flush them to disk when the memory
 // limit is exceeded.
-type transactionContainer struct {
-	transactions               []Transaction
-	currentMemSizeInBytes      int
-	maxMemSizeInBytes          int
-	flushToStorageRatio        float64
-	dropPrioritySorter         transactionPrioritySorter
-	optionalTransactionStorage transactionStorage
-	telemetry                  transactionContainerTelemetry
-	mutex                      sync.RWMutex
+type TransactionRetryQueue struct {
```

Comment:
Since ti's in the `retry` package we could rename this `TransactionQueue` or just `Queue`.

So the usage is `retry.Queue{}`.

Reply:
A type `Queue`, even in a package like `pkg/forwarder/internal/retry`, let me think it is a queue as defined by https://en.wikipedia.org/wiki/Queue_(abstract_data_type). This class is quite different.

Reply:
Well it seems to be a queue where we push transaction to be retried and pop some when we want to retry them, no ?

Just saying that `retry.TransactionRetryQueue` repeat the package name for no reason (see `Naming package contents
` in https://blog.golang.org/package-names).

Reply:
It is more than a simple `Queue`:
```
TransactionRetryQueue stores transactions in memory and flush them to disk when the memory
limit is exceeded
```

[ParseHTTPVersion](https://golang.org/pkg/net/http/#ParseHTTPVersion) repeats the name of the package (http).

Reply:
Of course it's more than a simple Queue, that's why it's not in `pkg/utils/` I think the internal implementation doesn't matter, it's a queue for transactions to retried.